internal/engine/wazevo/frontend/frontend.go

```diff
@@ -354,3 +362,42 @@ func SignatureForListener(wasmSig *wasm.FunctionType) (*ssa.Signature, *ssa.Sign
 	}
 	return beforeSig, afterSig
 }
+
+// isBoundSafe returns true if the given value is known to be safe to access up to the given bound.
+func (c *Compiler) getKnownSafeBound(v ssa.ValueID) *knownSafeBound {
+	if int(v) >= len(c.knownSafeBounds) {
+		return nil
+	}
+	return &c.knownSafeBounds[v]
+}
+
+// recordKnownSafeBound records the given safe bound for the given value.
+func (c *Compiler) recordKnownSafeBound(v ssa.ValueID, safeBound uint64, absoluteAddr ssa.Value) {
+	if int(v) >= len(c.knownSafeBounds) {
+		c.knownSafeBounds = append(c.knownSafeBounds, make([]knownSafeBound, v+1)...)
```

Comment:
This works, but creates a potentially much larger than need slice.

Alternatives:
```
if diff := int(v) - len(c.knownSafeBounds); diff >= 0 {
		c.knownSafeBounds = append(c.knownSafeBounds, make([]knownSafeBound, diff+1)...)
}
```

```
for int(v) >= len(c.knownSafeBounds) {
		c.knownSafeBounds = append(c.knownSafeBounds, knownSafeBound{})
}
```

Reply:
oh well this is intentional as this is repeatedly used over a module and we allocate the slice very generously